### DISCUSSION TITLE
Cannot stringify arbitrary non-POJOs

### DISCUSSION THREAD

**Discussion by SulaimanBarry** [nuxt/nuxt#25372] [Questions]
I am creating an application using Nuxt 3. I came across this error ("Cannot stringify arbitrary non-POJOs") after creating a middleware that gets the login user from sidebase useAuth and checks if the user provided their phone number with they signed in with Google for the first time. below is the code I have in my middleware. How can I fix this problem?
middleware code
`export default defineNuxtRouteMiddleware(async (to, from) => {
  const { status } = useAuth()
  if (status.value == 'authenticated') {
    const { data } = await useFetch('/api/profile/getUserOnly')
    console.log('heee', data.value)
    if (
      (data.value?.contact == null || data.value.contact == '') &&
      to.path != '/profile/need-info'
    ) {
      return navigateTo('/profile/need-info')
    }
  }
})
`
api/profile//getUserOnly code 
`
router.get(
  '/getUserOnly',
  defineEventHandler(async (event) => {
    const session = (await getServerSession(event)) as any
    const id = session?.user?.id as any
    const data = await db.query.users.findFirst({
      where: eq(users.id, id),
    })

    return {
      data,
    }
  }),
)

`

**Comment by TheAlexLichter** [comment]
don't return a class 😋 only provide plain objects as return value
